samples/snippets/src/main/java/com/example/spanner/admin/generated/CreateInstanceConfigSample.java

```diff
@@ -0,0 +1,86 @@
+ */
+
+package com.example.spanner.admin.generated;
+
+// [START spanner_create_instance_config]
+
+import com.google.cloud.spanner.admin.instance.v1.InstanceAdminClient;
+import com.google.spanner.admin.instance.v1.CreateInstanceConfigRequest;
+import com.google.spanner.admin.instance.v1.InstanceConfig;
+import com.google.spanner.admin.instance.v1.InstanceConfigName;
+import com.google.spanner.admin.instance.v1.ProjectName;
+import com.google.spanner.admin.instance.v1.ReplicaInfo;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class CreateInstanceConfigSample {
+
+  static void createInstanceConfig() throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String baseInstanceConfigId = "nam11";
+    String instanceConfigId = "custom-instance-config4";
+
+    createInstanceConfig(projectId, baseInstanceConfigId, instanceConfigId);
+  }
+
+  static void createInstanceConfig(
+      String projectId, String baseInstanceConfigId, String instanceConfigId) throws IOException {
+    InstanceAdminClient instanceAdminClient = InstanceAdminClient.create();
+    final InstanceConfigName baseInstanceConfigName = InstanceConfigName.of(projectId,
+        baseInstanceConfigId);
+    final InstanceConfig baseConfig =
+        instanceAdminClient.getInstanceConfig(baseInstanceConfigName.toString());
+    final InstanceConfigName instanceConfigName = InstanceConfigName.of(projectId,
+        instanceConfigId);
+    /**
+     * The replicas for the custom instance configuration must include all the replicas of the base
+     * configuration, in addition to at least one from the list of optional replicas of the base
+     * configuration.
+     */
+    final List<ReplicaInfo> replicas =
+        Stream.concat(baseConfig.getReplicasList().stream(),
+            baseConfig.getOptionalReplicasList().stream().limit(1)).collect(Collectors.toList());
+    final InstanceConfig instanceConfig =
+        InstanceConfig.newBuilder().setName(instanceConfigName.toString())
+            .setBaseConfig(baseInstanceConfigName.toString())
+            .setDisplayName("Instance Configuration").addAllReplicas(replicas).build();
+    final CreateInstanceConfigRequest createInstanceConfigRequest =
+        CreateInstanceConfigRequest.newBuilder().setParent(ProjectName.of(projectId).toString())
+            .setInstanceConfigId(instanceConfigId).setInstanceConfig(instanceConfig).build();
+    try {
+      System.out.printf("Waiting for create operation for %s to complete...\n", instanceConfigName);
+      InstanceConfig instanceConfigResult =
+          instanceAdminClient.createInstanceConfigAsync(
+              createInstanceConfigRequest).get(5, TimeUnit.MINUTES);
```

Comment:
Any significance of explicitly calling out 5 min for this LRO? This LRO (by default) has a total timeout of 5 minutes as specified here: https://github.com/googleapis/java-spanner/blob/58f94b200276d879f83e4432716b49baf3206226/google-cloud-spanner/src/main/java/com/google/cloud/spanner/admin/instance/v1/stub/InstanceAdminStubSettings.java#L916-L925

I think it would be beneficial to add a comment describing the underlying LRO behavior, but I don't think you need to explicitly call with `.get(5min)`.

Reply:
The code you have linked is for `RetrySetting`. A `get()` invocation on a `Future` object is a different thing. By default, the `get()` will block infinitely. And so it is preferred to specify a timeout.